### ISSUE TITLE
GitHub and GitLab Configuration Failure

### ISSUE THREAD

**Issue by mydiemho** [jhipster/jhipster-online#200]
After configure JHipster to access my GitHub account, I keep seeing the error `JHipster has failed to reach your Github repositories.`.  When I checked on GitHub, I can see that `JHipster` is in the `Authorized OAuth Apps` list.

![Screen Shot 2020-07-24 at 17 01 28](https://user-images.githubusercontent.com/1634185/88443988-b8c40580-cdcf-11ea-9d9c-ffa15891702b.png)

![Screen Shot 2020-07-24 at 17 01 50](https://user-images.githubusercontent.com/1634185/88443987-b82b6f00-cdcf-11ea-8380-6539b9bdf32a.png)

**Comment by SudharakaP** [comment]
@mydiemho : Thanks for reporting this. I was able to reproduce this. Although the JHipster app is authorized it shows this error it seems. Will take a look. :thinking:

**Comment by SudharakaP** [comment]
Just for the information; as pointed out in this comment; https://github.com/jhipster/jhipster-online/issues/93#issuecomment-663868193, this also applies to GitLab.

**Comment by vishal423** [comment]
@SudharakaP, what error do you see in server logs?

**Comment by SudharakaP** [comment]
@vishal423 : The error is,

<details>
<summary>
Server Log
</summary> 

```
A 2020-07-25T05:57:49Z Jul 25 05:57:49 jhipster-app jhonline.war[11687]: org.springframework.web.client.HttpClientErrorException$NotFound: 404 Not Found: [{"error":"Not Found"}] 
A 2020-07-25T05:57:49Z Jul 25 05:57:49 jhipster-app jhonline.war[11687]: #011at org.springframework.web.client.HttpClientErrorException.create(HttpClientErrorException.java:113) 
A 2020-07-25T05:57:49Z Jul 25 05:57:49 jhipster-app jhonline.war[11687]: #011at org.springframework.web.client.DefaultResponseErrorHandler.handleError(DefaultResponseErrorHandler.java:170) 
A 2020-07-25T05:57:49Z Jul 25 05:57:49 jhipster-app jhonline.war[11687]: #011at org.springframework.web.client.DefaultResponseErrorHandler.handleError(DefaultResponseErrorHandler.java:112) 
A 2020-07-25T05:57:49Z Jul 25 05:57:49 jhipster-app jhonline.war[11687]: #011at org.springframework.web.client.ResponseErrorHandler.handleError(ResponseErrorHandler.java:63) 
A 2020-07-25T05:57:49Z Jul 25 05:57:49 jhipster-app jhonline.war[11687]: #011at org.springframework.web.client.RestTemplate.handleResponse(RestTemplate.java:782) 
A 2020-07-25T05:57:49Z Jul 25 05:57:49 jhipster-app jhonline.war[11687]: #011at org.springframework.web.client.RestTemplate.doExecute(RestTemplate.java:740) 
A 2020-07-25T05:57:49Z Jul 25 05:57:49 jhipster-app jhonline.war[11687]: #011at org.springframework.web.client.RestTemplate.execute(RestTemplate.java:674) 
A 2020-07-25T05:57:49Z Jul 25 05:57:49 jhipster-app jhonline.war[11687]: #011at org.springframework.web.client.RestTemplate.postForEntity(RestTemplate.java:449) 
A 2020-07-25T05:57:49Z Jul 25 05:57:49 jhipster-app jhonline.war[11687]: #011at io.github.jhipster.online.web.rest.GitResource.saveToken(GitResource.java:126) 
A 2020-07-25T05:57:49Z Jul 25 05:57:49 jhipster-app jhonline.war[11687]: #011at io.github.jhipster.online.web.rest.GitResource$$FastClassBySpringCGLIB$$ca87e103.invoke(<generated>) 
A 2020-07-25T05:57:49Z Jul 25 05:57:49 jhipster-app jhonline.war[11687]: #011at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218) 
A 2020-07-25T05:57:49Z Jul 25 05:57:49 jhipster-app jhonline.war[11687]: #011at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:769) 
A 2020-07-25T05:57:49Z Jul 25 05:57:49 jhipster-app jhonline.war[11687]: #011at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163) 
A 2020-07-25T05:57:49Z Jul 25 05:57:49 jhipster-app jhonline.war[11687]: #011at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:747) 
```
</details>

**Comment by vishal423** [comment]
I did a quick check on local setup, and able to configure `GitHub` without any issues or additional configurations. Have you tried on local setup to reproduce the issue?

**Comment by SudharakaP** [comment]
> I did a quick check on local setup, and able to configure `GitHub` without any issues or additional configurations. Have you tried on local setup to reproduce the issue?

@vishal423 : That's strange. I created my own OAuth App and OAuth Token and tested it on local and I could recreate this exact scenario. :wink: :thinking: :thinking:

**Comment by vishal423** [comment]
![image](https://user-images.githubusercontent.com/15997681/88464703-7b986a00-ceda-11ea-93fa-75436ac13e3c.png)

**Comment by SudharakaP** [comment]
@vishal423 : Hmm.... :thinking: :thinking: Kinda confusing why this happens on some machines, 

![Screenshot from 2020-07-25 12-40-37](https://user-images.githubusercontent.com/12435965/88465002-275a9f00-ce74-11ea-88ab-d7d04e700a9a.png)

**Comment by SudharakaP** [comment]
@vishal423 : can you try running the prod profile and see if it works? :thinking:

**Comment by SudharakaP** [comment]
@vishal423 : For me it fails for dev as well. :disappointed:

**Comment by SudharakaP** [comment]
It seems to me that this intermittent issue of Git Sync on both GitHub and GitLab comes from way we send the POST request to the Git provider using the `RestTemplate` library. I cannot pinpoint why it fails only for some people but I felt that its something to do with the library itself so I changed it to the more recent `HttpClient` of Java 11. And thus the project was also upgraded to Java 11. Please take a look and let me know if you see any issues; I tested this on my side several times to make sure and seems everything works fine, but given that it wasn't working for some people I would appreciate if you can test it on your end as well if the sync is working as expected. :smile: 

cc: @vishal423

**Comment by SudharakaP** [comment]
@mydiemho : I just released v2.6.4 which includes the fix for this. If you even encounter this issue again please feel free to let us know. :smile: 

cc: @tonsV2

Thanks both of you for reporting this issue. :smile:

**Comment by devysf** [comment]
Hi, I am encountering this problem.
<img width="1645" alt="image" src="https://github.com/jhipster/jhipster-online/assets/40598633/cf7c7eaa-b557-4067-98be-2926632e43dc">


Additionally, although I receive access tokens on GitHub, I cannot list or select repositories.
<img width="1263" alt="image" src="https://github.com/jhipster/jhipster-online/assets/40598633/fc003786-659e-4b25-8523-1bbf0377209c">
